fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/IndexState.java

```diff
@@ -55,7 +55,15 @@ public enum IndexState {
      * reads or queries and it is not maintained
      * by the record store.
      */
-    DISABLED(2L, "indexesDisabled");
+    DISABLED(2L, "indexesDisabled"),
+    /**
+     * Indicates that this unique index is fully "indexed", but
+     * some uniqueness violations still exist. This may happen
+     * when the online indexer finds some duplicating records.
+     * In this mode, it is safe to consider an index as {@link #READABLE}
+     * for queries as long as uniqueness is not assumed.
+     */
+    READABLE_UNIQUE_PENDING(3L, "indexesReadableUniquePending");
```

Comment:
Would it make sense to be forward-looking and have this be `READABLE_CONSTRAINT_PENDING` to support potentially other types of constraints on indexes in the future?

Reply:
FMHO, uniqueness pending should be distinguished from other constraints - as it provides the specific information of an index being built but not unique. The planner, for Example, can benefit from having this specific info by the index state.